internal/gitlab/gitlab.go

```diff
@@ -383,21 +383,27 @@ func MRCreateNote(project string, mrNum int, opts *gitlab.CreateMergeRequestNote
 	return fmt.Sprintf("%s/merge_requests/%d#note_%d", p.WebURL, note.NoteableIID, note.ID), nil
 }
 
-// MRGet retrieves the merge request from GitLab project
-func MRGet(project string, mrNum int) (*gitlab.MergeRequest, error) {
-	p, err := FindProject(project)
+// MRGet retrieves the merge request from GitLab project ID (skipping the project lookup)
+func MRGetFromProjectID(projectID interface{}, mrNum int) (*gitlab.MergeRequest, error) {
+	mr, _, err := lab.MergeRequests.GetMergeRequest(projectID, mrNum, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	mr, _, err := lab.MergeRequests.GetMergeRequest(p.ID, mrNum, nil)
+	return mr, nil
+}
+
+// MRGet retrieves the merge request from GitLab project
+func MRGet(project string, mrNum int) (*gitlab.MergeRequest, error) {
+	p, err := FindProject(project)
 	if err != nil {
 		return nil, err
 	}
 
-	return mr, nil
+	return MRGetFromProjectID(p.ID, mrNum)
```

Comment:
@prarit , I think we can have a single function for getting the project object: the `interface{}` type here is used because it can be either `string` or `int`, it's handled in the go-gitlab module, since the [API](https://docs.gitlab.com/ee/api/projects.html#get-single-project) accepts both (ID as int and escaped project URL path as string).

With that, I think we can reduce the project lookup significantly. But, at the same time, we can leave this "optimization" for a future MR. Your call :D

Reply:
Oh, I didn't realize that about the call.  Let me remove it for now.

Reply:
So I fixed this and tested it with podman.  See changes.

Reply:
Yep.. Looks good to me. Tested it locally and seems to be working properly as well.